runtime/src/iree/base/internal/cpu.c

```diff
@@ -310,6 +310,74 @@ static void iree_cpu_initialize_from_platform_x86_64(uint64_t* out_fields) {
   out_fields[0] = out0;
 }
 
+#elif defined(IREE_ARCH_RISCV_64)
+#if defined(IREE_PLATFORM_ANDROID) || defined(IREE_PLATFORM_LINUX)
+
+// First 4 characters are "rv32" or "rv64"
+#define IREE_RISCV_64_ISA_SEARCH_START 4
+
+static void iree_cpu_initialize_from_platform_riscv_64(uint64_t* out_fields) {
+  char* line;
+  char isa[1024];
+  char* cpuinfo_path = "/proc/cpuinfo";
+  FILE* fp;
+  size_t len = 0;
+  ssize_t nread;
+
+  fp = fopen(cpuinfo_path, "r");
+  if (!fp) {
+    return;
+  }
+
+  while ((nread = getline(&line, &len, fp)) != -1) {
+    if (sscanf(line, "isa : %s\n", &isa[0])) {
+      // First 4 characters are "rv32" or "rv64"
+      // We search the character 'v' from the 4th character to '_' occur or
+      // string end.
+      char* first = strchr(&isa[0], '_');
+
+      // Can't find '_', search 'v' in the whole isa string.
+      if (!first) {
+        if (strchr(&isa[IREE_RISCV_64_ISA_SEARCH_START], 'v')) {
+          IREE_COPY_BITS(out_fields[0], IREE_CPU_DATA0_RISCV_64_RVV, 1, 1 << 0);
+        }
+        break;
+      } else {
```

Comment:
To avoid this double-negation (this else case is if the character `_` was NOT NOT found), flip the condition of this `if`:  `if(first) {...} else {...}`.